File: skill_sdk/tests/fixtures/js-project/src/core/TransformData.js
/**
 * Applies transformation rules to raw data. Mirrors TransformData in the Java fixture.
 */
export class TransformData {
  constructor() {
    throw new Error('Utility class — do not instantiate');
  }

  /**
   * @param {string} data
   * @returns {string}
   */
  static transform(data) {
    return `processed: ${data}`;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/core/TransformData.java
package com.fixture.core;

/**
 * Applies transformation rules to raw data. Mirrors TransformData function in main.go.
 */
public final class TransformData {

    private TransformData() {
    }

    public static String transform(String data) {
        return "processed: " + data;
    }
}
